cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -320,6 +320,30 @@ List<ProgramHistory> getRuns(Collection<ProgramReference> programs, ProgramRunSt
    */
   Map<ProgramRunId, RunRecordDetail> getActiveRuns(ProgramId programId);
 
+  /**
+   * Fetches the number of active (i.e PENDING or STARTING or RUNNING or SUSPENDED or STOPPING)
+   * run records against a given {@link ProgramReference} across versions of the program.
+   * If the number of active runs is higher than upperBound, then the upperBound value is returned.
+   *
+   * @param programRef the {@link ProgramReference} to match against
+   * @param upperBound int value acting as an upper bound for the active runs count, useful when we
+   *     only want to check if the number of active runs is lower than the upper bound. It ensures
+   *     that we do not scan more than upperBound number of active run records.
+   * @return int number of logged runs
+   */
+  int getProgramActiveRunsCount(ProgramReference programRef, int upperBound);
```

Comment:
we usually use term `limit` instead of `upperBound`.

Reply:
Done